.github/workflows/ci.yml

```diff
@@ -10,18 +10,63 @@ jobs:
   check:
     runs-on: ubuntu-22.04
 
+    services:
+      postgres:
+        image: postgis/postgis
+        env:
+          POSTGRES_USER: geoengine
+          POSTGRES_PASSWORD: geoengine
+          POSTGRES_DB: geoengine
+        ports:
+          - 5432:5432
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
     strategy:
       fail-fast: false
       matrix:
         # use all supported versions from https://devguide.python.org/versions/
         python-version: ["3.9", "3.10", "3.11", "3.12"]
 
+    defaults:
+      run:
+        working-directory: library
+
     steps:
-      - uses: actions/checkout@v3
-      - name: APT update
-        run: sudo apt-get update
-      - name: Install system dependencies
-        run: sudo apt-get install libgeos-dev libproj-dev
+      - name: Checkout library code
+        uses: actions/checkout@v4
+        with:
+          path: library
+      - name: Read backend version
+        id: read-backend-version
+        run: echo "GEOENGINE_VERSION=$(cat .ci_instance)" >> $GITHUB_OUTPUT
```

Comment:
- could this be read from/encoded in the openapi-client? we already specify this in the setup.cfg. On the other hand we mabye do not want to release a new openapi-client when we change something in the backend that does not affect the API...
- would it make sense to have the compatible geo engine version available in the python code to check compatibility with of the library with the backend
- in any case: I think the name of the .ci_instance is not ideal. maybe name it .backend_commit_hash and place it in the .github/worklfows folder?

Reply:
> ould this be read from/encoded in the openapi-client? 
Ne, dafür müssten wir richtig versionieren.

> would it make sense to have the compatible geo engine version available in the python code to check compatibility with of the library with the backend
Wie genau? Geht ja meistens auch mit einer anderen Version

[30073a4](https://github.com/geo-engine/geoengine-python/pull/211/commits/30073a42a7c434fa8dd1b707ee184d8f8c80f4ad)